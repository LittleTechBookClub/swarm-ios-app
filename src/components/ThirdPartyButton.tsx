import React from 'react';
import { StyleSheet, TouchableOpacity, Image } from 'react-native';

type Props = {
  imgSrc: string;
  handleClick: () => void;
  width: number;
  height: number;
};

const ThirdPartyButton: React.FC<Props> = ({ imgSrc, handleClick, width, height }: Props) => {
  return (
    <TouchableOpacity onPress={handleClick} style={styles.buttonContainer}>
      <Image source={{ uri: imgSrc }} style={{ width: width, height: height }} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    width: '30%',
    height: 56,
    borderRadius: 8,
    borderColor: '#E8ECF4',
    paddingVertical: 16,
    paddingHorizontal: 40,
    borderWidth: 1,
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: 'center',
  },
});
export default ThirdPartyButton;
