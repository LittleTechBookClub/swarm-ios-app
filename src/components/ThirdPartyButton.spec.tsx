/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import ThirdPartyButton from './ThirdPartyButton';

describe('ThirdPartyButton', () => {
  it('should render the Third Party Button', () => {
    const tree = render(
      <ThirdPartyButton imgSrc="test" handleClick={() => {}} width={20} height={20} />
    );
    expect(tree).toMatchSnapshot();
  });
});
