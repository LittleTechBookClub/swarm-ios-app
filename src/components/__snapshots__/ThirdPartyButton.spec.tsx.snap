// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdPartyButton should render the Third Party Button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#E8ECF4",
      "borderRadius": 8,
      "borderWidth": 1,
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 40,
      "paddingVertical": 16,
      "textAlign": "center",
      "width": "30%",
    }
  }
>
  <Image
    source={
      {
        "uri": "test",
      }
    }
    style={
      {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
